reconciler/configstore.go

```diff
@@ -16,10 +16,26 @@ limitations under the License.
 
 package reconciler
 
-import "context"
+import (
+	"context"
+
+	"k8s.io/apimachinery/pkg/types"
+)
 
 // ConfigStore is used to attach the frozen configuration to the context.
 type ConfigStore interface {
 	// ConfigStore is used to attach the frozen configuration to the context.
 	ToContext(ctx context.Context) context.Context
 }
+
+// OnDeletionInterface defines the strongly typed interface to be implemented by a
+// controller observing a deletion of {{.type|raw}}. Every controller that was active
+// during the deletion of the respective resource is guaranteed to observe this event,
+// leader or not. It's usually used to clear up in-memory state regarding the respective
+// resource. Finalizers should be used to ensure external resources are properly cleaned
+// up.
+type OnDeletionInterface interface {
+	// ObserveDeletion implements custom logic to observe deletion of the respective resource
+	// with the given key.
+	ObserveDeletion(ctx context.Context, key types.NamespacedName) error
```

Comment:
opted for returning `error` vs. `reconciler.Event` due to the unavailablity of the entire `runtime.Object` on this path. Open to suggestions.

Reply:
Observers should not return anything seems like?

Reply:
why is this in config store?